src/components/ControlledAddressCombobox.tsx

```diff
@@ -0,0 +1,87 @@
+import { Combobox, Transition } from "@headlessui/react";
+import { Control, Controller, FieldError } from "react-hook-form";
+import { OnboardingFormInputs } from "../pages/profile";
+import { Fragment, SetStateAction } from "react";
+import { CarpoolFeature, CarpoolAddress } from "../utils/types";
+
+interface ControlledAddressComboboxProps {
+  control: Control<OnboardingFormInputs>;
+  name: "startAddress" | "companyAddress";
+  addressSelected: CarpoolAddress;
+  addressSetter: (val: SetStateAction<CarpoolAddress>) => void;
+  addressUpdater: (val: SetStateAction<string>) => void;
+  addressSuggestions: CarpoolFeature[];
+  error?: FieldError;
+}
+
+const ControlledAddressCombobox = (props: ControlledAddressComboboxProps) => {
+  return (
+    <Controller
+      name={props.name}
+      control={props.control}
+      render={({ field: { ref, ...fieldProps } }) => (
+        <Combobox
+          className={`w-full`}
+          as="div"
+          value={props.addressSelected}
+          onChange={(val: CarpoolFeature) => {
```

Comment:
What's the difference between the onChange for the Combobox and the onChange for Combobox.Input?

Reply:
onChange for ComboBox.Input runs everytime that the user makes any change in the input field. onChange for Combobox only runs when the user actually clicks on one of the addresses and chooses it. There is some possibility that they can possibly reduced to only one of them, but will need to look into it.